homeassistant/components/sensor/seventeentrack.py

```diff
@@ -0,0 +1,236 @@
+        login_result = await client.profile.login(
+            config[CONF_USERNAME], config[CONF_PASSWORD])
+
+        if login_result is False:
+            _LOGGER.error('Invalid username and password provided')
+            return False
+    except SeventeenTrackError as err:
+        _LOGGER.error('There was an error while logging in: %s', err)
+        return False
+
+    if CONF_SCAN_INTERVAL in config:
+        scan_interval = config[CONF_SCAN_INTERVAL]
+    else:
+        scan_interval = DEFAULT_SCAN_INTERVAL
+
+    data = SeventeenTrackData(
+        client, scan_interval, config[CONF_SHOW_ARCHIVED],
+        config[CONF_SHOW_DELIVERED])
+    await data.async_update()
+
+    sensors = []
+
+    for status, quantity in data.summary.items():
+        sensors.append(SeventeenTrackSummarySensor(data, status, quantity))
+
+    for package in data.packages:
+        sensors.append(SeventeenTrackPackageSensor(data, package))
+
+    async_add_entities(sensors, True)
+
+
+class SeventeenTrackSummarySensor(Entity):
+    """Define a summary sensor."""
+
+    def __init__(self, data, name, initial_state):
+        """Initialize."""
+        self._attrs = {ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION}
+        self._data = data
+        self._name = name
+        self._state = initial_state
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        return self._attrs
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return 'mdi:package'
+
+    @property
+    def name(self):
+        """Return the name."""
+        return 'Packages {0}'.format(self._name)
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def unique_id(self):
+        """Return a unique, HASS-friendly identifier for this entity."""
+        return '{0}_{1}'.format(self._data.account_id, slugify(self._name))
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        if self._state == 1:
+            return 'package'
+        return 'packages'
+
+    async def async_update(self):
+        """Update the sensor."""
+        await self._data.async_update()
+
+        self._state = self._data.summary[self._name]
+
+
+class SeventeenTrackPackageSensor(Entity):
+    """Define an individual package sensor."""
+
+    def __init__(self, data, package):
+        """Initialize."""
+        self._attrs = {
+            ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION,
+            ATTR_DESTINATION_COUNTRY: package.destination_country,
+            ATTR_INFO_TEXT: package.info_text,
+            ATTR_LOCATION: package.location,
+            ATTR_ORIGIN_COUNTRY: package.origin_country,
+            ATTR_PACKAGE_TYPE: package.package_type,
+            ATTR_TRACKING_INFO_LANGUAGE: package.tracking_info_language,
+        }
+        self._data = data
+        self._name = package.tracking_number
```

Comment:
So these entities for package tracking are only added once when home assistant starts and then after they are delivered they are never updated again?

Reply:
Hmm, this is a great point: although "Delivered" packages will be potentially ignored on HASS start (based on configuration), as soon as a package becomes "Delivered", that entity will remain. Not a great user experience. Suggestions? Is there a nice, approved way to periodically scrub these dead entities out (again assuming that's in configuration)?

Reply:
Adding new entities for new packages that should be tracked is straight forward, just keep a reference to `async_add_entities`, and add entities as needed after each new scan.

Removing is not hard per se, we have entity method, to do so, but question is when should that be done. There is some value in knowing that the package reached delivered status, right? So we can't remove immediately when that happens.

Reply:
Right.

@tjorim, you've been following this one – what would you ideally see happen? Wait some configurable time before removing delivered packages?

Reply:
What about creating a persistent notification once it's delivered? Then the user knows about it and the entity can be removed.
It's not ideal since the history is gone, but good enough since users just have this for tracking. (Maybe a link to the site to track that specific package.)

Reply:
Ooh, that’s interesting. @MartinHjelmare, opinion?

Reply:
Are packages ever cleared from the 17track api? An alternative could be to just return the result from the api, and remove the entity when the package is no longer present in the result.

Notification could work. We fire a state change event when removing the entity, so automations are possible too.